scripts/release.sh

```diff
@@ -337,19 +337,26 @@ main() {
     log_warning "WARNING: If not running on DRY_MODE, please do the GitHub release manually."
     log_warning ""
   else
+    read -p "Create GitHub draft release in ${REPOSITORY} [y/N]? " -r confirm
+    [[ "${confirm,,}" == "y" ]] || exit 1
+
     local gh_repo
     local release_notes_temp_file
     local release_url
     local gh_release_args=()
 
-    # For the main branch (v3.6), we should mark the release as a prerelease.
-    # The release-3.5 (v3.5) branch, should be marked as latest. And release-3.4 (v3.4)
-    # should be left without any additional mark (therefore, it doesn't need a special argument).
-    if [ "${BRANCH}" = "main" ]; then
-      gh_release_args=(--prerelease)
-    elif [ "${BRANCH}" = "release-3.5" ]; then
-      gh_release_args=(--latest)
-    fi
+    # Explicitly set release arguments per branch.
+    case "${BRANCH}" in
+      "main")
+        gh_release_args=(--latest=false --prerelease=true)
+        ;;
+      "release-3.5")
```

Comment:
We might want to start capturing a list of tasks in an issue and adding it to the https://github.com/etcd-io/etcd/milestone/38 in order to capture the trivial release related tasks we would need to perform in order to safely cut new `release-3.6` branch. For example, updating this script to add the `release-3.6` logic.